conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanGenerator.java

```diff
@@ -126,7 +126,8 @@ public static JavaFile generateBeanType(
                 .addAnnotations(safety)
                 .build());
 
-        if (poetFields.size() <= MAX_NUM_PARAMS_FOR_FACTORY) {
+        if (poetFields.isEmpty()
```

Comment:
Checkstyle is complaining about the cyclomatic complexity here. Will see if there is a nice refactor.

Reply:
Opened #2088.